django/db/backends/ddl_references.py

```diff
@@ -198,3 +198,49 @@ def rename_column_references(self, table, old_column, new_column):
 
     def __str__(self):
         return self.template % self.parts
+
+
+class Expressions(Reference):
+    def __init__(self, table, expressions, compiler, quote_value, opclasses):
+        self.table = table
+        self.expressions = expressions
+        self.compiler = compiler
+        self.quote_value = quote_value
+        self.opclasses = opclasses
+
+    def references_table(self, table):
+        return self.table == table
+
+    def references_column(self, table, column):
+        columns = (col.target.column for col in self.compiler.query._gen_cols(self.expressions))
+        return self.table == table and column in columns
+
+    def rename_table_references(self, old_table, new_table):
+        if self.table == old_table:
+            self.table = new_table
+
+    def rename_column_references(self, table, old_column, new_column):
+        if self.table == table:
+            cols = self.compiler.query._gen_cols(self.expressions)
+            for index, col in enumerate(cols):
+                if col.target.column == old_column:
+                    # TODO: set_source_expressions somehow, or can we mutable col.target.column?
+                    self.expressions[index] = col
```

Comment:
@hannseman I see what you mean! Going through the `set_source_expressions` route would be very hard as you'd have to set it for all the nested expression leading to the `Col`. I guess `copy()`'ing all expressions preemptively would be best way forward

```suggestion
        if self.table != table:
            return
        renamed_expressions = []
        for expression in enumerate(self.expressions):
            renamed_expression = expression.copy()
            for col in self.compiler.query._gen_cols(renamed_expression):
                if col.target.column == old_column:
                    col.target.column = new_column
            renamed_expressions.append(renamed_expression)
        self.expressions = renamed_expressions
```

Reply:
Many thanks! That makes sense, had to use deepcopy. 

I've added a test case for the `Expressions` class.